### PR TITLE
Create new namespace "Domain" for Domain Driven Design classes

### PR DESCRIPTION
**Pull request by nupplaphil**
At the hackathon, I'd like to discuss more about "DDD" and the reason behind it.

Therefore I'd like to cleanup a little bit the structure before.
Everything "DDD" relevant is now moved under the new `Domain` namespace.

As you now can see, not every factory, model or object is moved there, which should indicate the difference between "classic" factory, model, objects classes and the DDD relevant ones.

We now can see/discuss better the patterns behind it.

Draft PR until @MrPetovan say it's OK for him :-) 

the namespace/directory structure:
- `Domain`
- `Domain\Collection`
- `Domain\Entity` (API\Mastodon, .. - instead of the `Object` namespace)
- `Domain\Factory`
- `Domain\Model`
- `Domain\Repository`

[edit] some bedtime reading - just as impression about the "why" and maybe a more practical point of view because of our long-time discussion about it ;-) : https://dev.to/ludofleury/domain-driven-design-with-php-and-symfony-1bl6

### CODE REVIEW

File: src/Module/Register.php [hunk 1]
```diff
@@ -367,18 +367,18 @@ public static function post(array $parameters = [])
...
-					'source_mail'  => $user['email'],
-					'source_nick'  => $user['nickname'],
-					'source_link'  => $base_url . '/admin/users/',
-					'link'         => $base_url . '/admin/users/',
-					'source_photo' => $base_url . '/photo/avatar/' . $user['uid'] . '.jpg',
-					'to_email'     => $admin['email'],
-					'uid'          => $admin['uid'],
-					'language'     => ($admin['language'] ?? '') ?: 'en',
-					'show_in_notification_page' => false
+								  'type'         => \Friendica\Domain\Model\Notify\Type::SYSTEM,
```
Comment (MrPetovan) [inline src/Module/Register.php]: Whoa, the indentation here is out of whack.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Comment by MrPetovan** [comment]
So, what are the non-static classes that didn't make the cut?

**Comment by annando** [comment]
@nupplaphil be prepared for long hours explanations at the hackathon, preferably in German when @MrPetovan is not around, since I don't understand the linked text from the beginning. I never heard of "Bounded Contexts", "Microservice", "CQRS", "Command handlers", "Event Sourcing", ... And what is a "Domain" in this context at all?

**Comment by annando** [comment]
Being a little bit more awake, I have some questions to the text and your implementation. I don't think that adding `/Domain` isn't in the intention of that text. The text describes the "Domain" as some kind of department/purpose separator and "Domain" should only be used when there are several. So - I think - we had to define which "Domains" we do have. And if I understood the text then the domain path should contain the domain description in it, like the example `/Domain/Legal/Company/France/RegistrationNumber/Siret.php`.

Having this in mind, I don't think that `\Friendica\Domain\Repository\Notify` and `Friendica\Domain\Model\Notify\Type` are the right names, because - I think - "Repository" and "Model" are no domains, but "Notify" possibly is?

Concerning the separation, I think that we can separate between backend, API, CLI and frontend. And the backend - since it is really large - should be separated in the protocols, the message processing, communication, ...

**Comment by nupplaphil** [comment]
I've to think more about it, but as far as I understood you @annando I fully agree and that's what I want to discuss with you. The domain namespace is therefore temporary to exactly discuss this type of re-organisatin away from namespace per class types to namespaces per domain :-)

And splitting by frontend, API and CLI is exactly what my Plans are from the beginning of the refactoring.. And I think I like the split based on protocols. Etc. , ... I think there we have our real domains found :-)

**Comment by nupplaphil** [comment]
Closing because it isn't a good idea..

**Comment by annando** [comment]
Have you abandon it completely or do you just need to refine it? I really would like to see some class approach that I am able to understand, I must confess :-) So some (online) meeting in the next time between (at least) you, @MrPetovan and me would sound cool to me.

**Comment by nupplaphil** [comment]
I'd like to take a step forward to "real" domains as discussed in the dev forum. This would mean not a separation by types (repo, model, ...) and instead a separation by functionality/domain (api/front-end/protocol and a layer under it the type differention). And each domain has low bounds inside but strict bounds outside to other domains

**Comment by MrPetovan** [comment]
What about the "common" classes that can be used by multiple endpoints like models?

**Pull request closed without merging**